docs/topics/object-declarations.md

```diff
@@ -87,32 +138,63 @@ In all these cases, members added in the anonymous object are not accessible. Ov
 are declared in the actual type of the function or property:
 
 ```kotlin
-interface A {
-    fun funFromA() {}
+//sampleStart
+interface Notification {
+    // Declares notifyUser() in the Notification interface
+    fun notifyUser()
 }
-interface B
 
-class C {
-    // The return type is Any; x is not accessible
-    fun getObject() = object {
-        val x: String = "x"
+interface DetailedNotification
+
+class NotificationManager {
+    // The return type is Any; message is not accessible
+    // When the return type is Any, only members of Any are accessible.
+    fun getNotification() = object {
+        val message: String = "General notification"
     }
 
-    // The return type is A; x is not accessible
-    fun getObjectA() = object: A {
-        override fun funFromA() {}
-        val x: String = "x"
+    // The return type is Notification because the anonymous object implements only one interface
+    // notifyUser() is accessible because it is part of Notification
+    // The message property is not accessible because it is not declared in the Notification interface
+    fun getEmailNotification() = object : Notification {
+        override fun notifyUser() {
+            println("Sending email notification")
+        }
+        val message: String = "You've got mail!"
     }
 
-    // The return type is B; funFromA() and x are not accessible
-    fun getObjectB(): B = object: A, B { // explicit return type is required
-        override fun funFromA() {}
-        val x: String = "x"
+    // The return type is DetailedNotification; notifyUser() and message are not accessible
+    // Only members declared in the DetailedNotification interface are accessible
+    fun getDetailedNotification(): DetailedNotification = object : Notification, DetailedNotification {
+        override fun notifyUser() {
+            println("Sending detailed notification")
+        }
+        val message: String = "Detailed message content"
     }
 }
+//sampleEnd
+fun main() {
+    val notificationManager = NotificationManager()
+    // No output
```

Comment:
```suggestion
```

Reply:
I think this is an important part to understand that these produce no output, how about moving the comment?
    // Produces no output
    val notificationManager = NotificationManager()

Reply:
Yes! Much better.